catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -355,9 +358,42 @@ private String getExceptionMessage(String responseType, String previousException
         return builder.toString();
     }
 
+    public void shutdown() {
+        getPeriodicDrain().shutdown();
```

Comment:
That is an ugly block of code. I can wonder why something like this isn't already part of the ExecutorService implementations? It is the kind of thing you want to hide away instead of instead of having ExecutorService clients do this. Have a look at item #3 in this article: http://www.nurkiewicz.com/2014/11/executorservice-10-tips-and-tricks.html. How about that approach?